subsys/bluetooth/host/conn.c

```diff
@@ -1803,19 +1803,24 @@ struct bt_conn *bt_conn_lookup_handle(u16_t handle)
 	return NULL;
 }
 
-int bt_conn_addr_le_cmp(const struct bt_conn *conn, const bt_addr_le_t *peer)
+int bt_conn_is_peer_addr_le(const struct bt_conn *conn, u8_t id,
```

Comment:
Looks like this is expecting a int as return but instead it return true/false.

Reply:
It might be cleaner if we have something like:

struct bt_le_peer {
  u8_t id;
  bt_addr_le_t *addr;
}

And then replace where we might need to store them both and we can call it bt_conn_is_le_peer and keep bt_conn_addr_le_cmp for just comparing the address.

Reply:
You could also consider adding a `u8_t id` parameter to `bt_conn_addr_le_cmp`, since I can't think of any use case where you would want to compare the address and ignore the identity.

Reply:
Fixed the return value.

The other change I'm not so sure about. Mostly because it seems like it would cause ripple effect to the functions. Most of the functions accept an ID and and address, and some of these functions are public API functions. In that case we would have to construct this object.

I added `u8_t` parameter to `bt_conn_addr_le_cmp` but since it does not strictly only return values from addr_cmp I changed it to return true/false and renamed it. So the old function is in use anymore. And I don't see the need to have it either.